src/rgw/rgw_req_context.h

```diff
@@ -0,0 +1,16 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "common/async/yield_context.h"
+#include "common/tracer.h"
+
+class DoutPrefixProvider;
+
+// this struct holds information which is created at the frontend
+// and should trickle down through all function calls to the backend
+struct req_context {
+  const DoutPrefixProvider* dpp{nullptr};
+  const optional_yield y{null_yield};
+  jspan span;
```

Comment:
could we change `add_span()` or add an overload that takes the raw `Span` pointer instead? storing a shared_ptr in `req_context` means that we have to do atomic refs every time we copy or construct a new one. using a raw pointer leaves ownership to the stack where it belongs

Reply:
see: https://github.com/ceph/ceph/pull/53547#discussion_r1332841614